src/Components/Firestore.js

```diff
@@ -303,3 +319,69 @@ export async function MarkNotificationsSeenOrRead(notiArray, IdToNotify, verb) {
     }
   }
 }
+
+export async function getRandomCommunityList() {
+  // Repeat call to try and ensure a list with at least (1) anime title is found.
+  for (let i = 1; i < 6; i++) {
+    try {
+      const randomNumber = Math.floor(Math.random() * 9999999);
+      const q = query(
+        collection(db, "watchlistData"),
+        where("random", ">=", randomNumber),
+        limit(1)
+      );
+      let querySnapshot = await getDocs(q);
+      let listInfo;
+      querySnapshot.forEach((doc) => {
+        listInfo = { ...doc.data() };
+      });
+      if (!listInfo) {
+        const wrapAroundQuery = query(
+          collection(db, "watchlistData"),
+          where("random", "<=", randomNumber),
+          limit(1)
+        );
+        querySnapshot = await getDocs(wrapAroundQuery);
+        querySnapshot.forEach((doc) => {
+          listInfo = { ...doc.data() };
+        });
+      }
+      // Populate anime info for list selected above
+      const docRef = doc(db, "users", listInfo.userId);
+      const data = await getDoc(docRef);
+      const lists = data.data()?.lists;
+      for (let item of lists) {
+        if (item.id === listInfo.listId && item.anime.length > 0) {
+          const data = {
+            userId: listInfo.userId,
+            listId: listInfo.listId,
+            // Only pull 24 shows from list, to prevent unnecessary firestore reads
+            anime: item.anime.slice(0, 24),
+          };
+          return data;
+        }
+      }
+    } catch (error) {
+      console.error(
+        "Error fetching random community list from Firestore",
+        error
+      );
+    }
+  }
+}
+
+export async function deleteWatchlistDataEntry(userId, listId) {
+  const listDocRef = doc(db, "watchlistData", userId + listId);
+
+  return runTransaction(db, async (transaction) => {
+    // This code may get re-run multiple times if there are conflicts.
+    const listDoc = await transaction.get(listDocRef);
+
+    // The deletion will only succeed if the document read above has not changed since then.
+    // Otherwise, it will re-run this function.
+    transaction.delete(listDocRef);
+  });
+  // To-do: Write script to delete all sub-collections WITHOUT an ancestor.
```

Comment:
I can do this.  Once that exists, we may want a 'deleted' flag that we can set on the watchlistData so it is easy to query just those that are marked for deletion, which will reduce unnecessary reads.

Reply:
Great thinking!!

I now only delete the `random` field so it won't get picked to show up on the home page, and I add `awaitingDeletion: true` to the doc.  Let me know if you'd like this flagging mechanism to look/act differently!

Reply:
Great improvements, that will work nicely.  One wipeout syncer coming right up...